tests/functional/test_useragent.py

```diff
@@ -0,0 +1,32 @@
+from awscli import __version__ as awscli_version
+from tests import CLIRunner
+
+def assert_user_agent_has_necessary_components(result, service, operation):
```

Comment:
We can probably shorten this name to `assert_expected_user_agent`. I like that because we can then potentially expand it to include expected values as part of the function signature for some of these components down the road (e.g. `expected_installer`, `expected_arch`, etc.)